src/ctia/entity/event/obj_to_event.clj

```diff
@@ -43,12 +46,49 @@
          (remove #(contains? diff-before %)
                  (keys diff-after))))))
 
+(defn update-changes
+  "Returns a list of changes for an Update event"
+  [old new]
+  {:pre [(map? old)
+         (map? new)]
+   :post [(vector? %)]}
+  (let [old-keys (set (keys old))
+        new-keys (set (keys new))
+        same-keys (set/intersection new-keys old-keys)
+        added-keys (set/difference new-keys old-keys)
+        deleted-keys (set/difference old-keys new-keys)]
+    (vec
+      ;; sort by :field then :action
+      (sort-by
+        (juxt :field :action)
+        (concat
+          (mapcat (fn [k]
+                    {:pre [(keyword? k)]}
+                    (let [[oldv newv] (map k [old new])]
+                      (when (not= oldv newv)
+                        [{:field k
+                          :action "modified"
+                          :change {:before oldv
+                                   :after newv}}])))
```

Comment:
shouldn't we handle nested changes?

Reply:
This PR adds `added` and `removed` events. `modified` events are left unchanged.

Reply:
https://github.com/threatgrid/iroh/issues/3551#issuecomment-635444332